mbed_targets/mbed_target_build_attributes.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+from dataclasses import dataclass
+from typing import FrozenSet, Dict
+
+from mbed_targets.exceptions import TargetBuildAttributesError
+from mbed_targets._internal import target_attributes
+
+
+@dataclass(frozen=True, order=True)
+class MbedTargetBuildAttributes:
+    """Representation of a target definition in Mbed OS library's targets.json file.
+
+    These attributes contains the specific information needed to build Mbed applications for this target.
+
+    Attributes:
+        labels: Sections of Mbed OS to be included in builds for this target.
+        features: Sections of Mbed OS features to be included in builds for this target.
+        components: Sections of Mbed OS components to be included in builds for this target.
+        config: Configuration defaults for this target to be used in builds.
+        supported_toolchains: Toolchains that can be used to build for this target.
+        default_toolchain: Default toolchain used for this target.
+    """
+
+    labels: FrozenSet[str]
+    features: FrozenSet[str]
+    components: FrozenSet[str]
+    config: Dict[str, str]
+    supported_toolchains: FrozenSet[str]
+    default_toolchain: str
```

Comment:
This should be `Optional[str]` as below `get('', None)` is used.

Reply:
Shows how good `mypy` is, eh? ;)

Reply:
efc3f58